pkg/volume/csi/csi_attacher_test.go

```diff
@@ -754,7 +754,7 @@ func TestAttacherWaitForVolumeAttachment(t *testing.T) {
 			if tc.shouldFail && err == nil {
 				t.Error("expecting failure, but err is nil")
 			}
-			if tc.initAttachErr != nil {
+			if tc.initAttachErr != nil && err != nil {
```

Comment:
from the test cases, I see if initAttachErr != nil,  then shouldFail = true, which means err must be !nil at this point. Otherwise, the check will fail above. So I don't see it is very necessary to change here unless you expect to have test case that initAttachErr != nil, and shouldFail = false.

Reply:
Yes, I didn't notice that the initAttachErr and shouldFail fields were corresponding :  `if initAttachErr != nil, then shouldFail = true`. Thanks !
But now this is the right change, easier to read the code.